File: api/src/main/java/com/api/app/emum/PAY001.java
package com.api.app.emum;

import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * 결제유형코드: payTypeCode
 * */
@Getter
@AllArgsConstructor
public enum PAY001 implements CommonCode {
    PAYMENT("001", "결제", 1, "", ""),
    REFUND("002", "환불", 2, "", ""),
    ;

    private final String code;
    private final String codeName;
    private final int displaySequence;
    private final String referenceValue1;
    private final String referenceValue2;

    public static PAY001 findByCode(String code) {
        return CommonCodeUtil.findByCode(PAY001.class, code);
    }
}

File: api/monolith/src/main/kotlin/com/api/app/emum/PAY001.kt
package com.api.app.emum

/** 결제구분코드: payTypeCode */
enum class PAY001(
    override val code: String,
    override val codeName: String,
    override val displaySequence: Int,
    override val referenceValue1: String,
    override val referenceValue2: String
) : CommonCode {
    PAYMENT("001", "결제", 1, "", ""),
    REFUND("002", "환불", 2, "", ""),
    ;

    companion object {
        fun findByCode(code: String): PAY001? = CommonCodeUtil.findByCode(PAY001::class.java, code)
    }
}
